poker-pals/src/components/pokerComponents/HandRankCalculator.vue

```diff
@@ -0,0 +1,701 @@
+        </button>
+
+        <button v-on:click="calculate()">get number</button>
+        <p id="result">0</p>
+        <p id="resultChars">0</p>
+    </div>
+</template>
+
+<style scoped>
+
+    .card {
+        float: left;
+        width: 71px;
+        height: 96px;
+        overflow: hidden;
+        opacity: 0.3;
+    }
+
+   .card img {
+        width: 923px;
+        height: 384px;
+    }
+
+    .box{
+        width: 923px;
+        margin: auto;
+    }
+
+</style>
+
+<script>
+import Card from './Card';
+
+export default {
+    name: "HandRankCalculator",
+    mounted(){
+        for(let i = 1; i <= 4; i++){
+            for(let j = 1; j <= 13; j++){
+                let a = -1 - (96 * (i-1));
+                let b = -9 - (71 * (j-1));
+                let div = document.getElementById(i.toString() + '_' + j.toString());
+                div.getElementsByTagName('img')[0].style.margin = a.toString() + "px 0px 0px " + b.toString() + "px";
+                //document.getElementById("myDiv").style.margin = "50px 10px 20px 30px"; 
+            }
+        }
+    },
+    methods:{
+        toggleImage(id){
+            let cardImage = document.getElementById(id);
+            //console.log(JSON.cardImage);
+            if(cardImage.style.opacity < 1.0){
+                cardImage.style.opacity = 1.0;
+            }
+            else{
+                cardImage.style.opacity = 0.3;
+            }
+        },
+        calculate(){
+            let cards = new Array(7);
+            let k = 0;
+            for(let i = 1; i <= 4; i++){
+                for(let j = 1; j <= 13; j++){
+                    let div = document.getElementById(i.toString() + '_' + j.toString());
+                    if(div.style.opacity > 0.3){
+                        // found a selected card
+                        cards[k] = new Card(j, Card.getSuit(i));
+                        k++;
+                        if(k >= 7){
+                            // break the loops
+                            i = 5;
+                            j = 14;
+                        }
+                    }
+                }
+            }
+            if(k < 7){
+                console.log('not enough cards');
+                return;
+            }
+
+            let result = -1;
+            
+            result = this.handStraightFlushValue(cards);
+            if(result != -1){
+                document.getElementById('result').innerHTML = result.toString(16);
+                document.getElementById('resultChars').innerHTML = this.convertToChars(result.toString(16));
+                return;
+            }
+            result = this.handQuadsValue(cards);
+            if(result != -1){
+                document.getElementById('result').innerHTML = result.toString(16);
+                document.getElementById('resultChars').innerHTML = this.convertToChars(result.toString(16));
+                return;
+            }
+            result = this.handFullHouseValue(cards);
```

Comment:
You've been copy/pasting this logic over and over again. Maybe extract it to a function to make the code look more clean.